### ISSUE TITLE
got 'nan' loss

### ISSUE THREAD

**Issue by denny1108** [daviddao/spatial-transformer-tensorflow#1]
Hi david, 

I ran the cluttered_mnist example based on your implementation of SPN. I used exactly the same data as you, and the same code. However, I got 'nan' loss somewhere during training. Do you have any idea about this. 

Thanks,

**Comment by daviddao** [comment]
Hi denny,

https://github.com/tensorflow/models/pull/181 solves the issue. 
tl;dr: need to replace my softmax with tf build-in softmax.

I will also merge the updated version of my STN in TensorFlow/models with this version soon.

**Comment by daviddao** [comment]
Close with 65a67c6
